src/SLCore.Listeners.UnitTests/CredentialsListenerTests.cs

```diff
@@ -0,0 +1,162 @@
+
+        response.Should().BeSameAs(GetCredentialsResponse.NoCredentials);
+    }
+
+    [TestMethod]
+    public async Task GetCredentialsAsync_SonarQubeCredentialsNotFound_ReturnsNoCredentials()
+    {
+        var testSubject = CreateTestSubject(out var credentialStoreMock);
+        credentialStoreMock.Setup(x => x.ReadCredentials(It.Is((TargetUri uri) => UriEquals(uri, SonarQubeUri)))).Returns((Credential)null);
+
+        var response = await testSubject.GetCredentialsAsync(new GetCredentialsParams(SonarQubeConnectionId));
+
+        response.Should().BeSameAs(GetCredentialsResponse.NoCredentials);
+    }
+    
+    [TestMethod]
+    public async Task GetCredentialsAsync_SonarCloudCredentialsNotFound_ReturnsNoCredentials()
+    {
+        var testSubject = CreateTestSubject(out var credentialStoreMock);
+        credentialStoreMock.Setup(x => x.ReadCredentials(It.Is((TargetUri uri) => UriEquals(uri, SonarCloudUri)))).Returns((Credential)null);
+
+        var response = await testSubject.GetCredentialsAsync(new GetCredentialsParams(SonarQubeConnectionId));
+
+        response.Should().BeSameAs(GetCredentialsResponse.NoCredentials);
+    }
+    
+    [TestMethod]
+    public async Task GetCredentialsAsync_SonarQubeUsernameAndPasswordFound_ReturnsUsernameAndPassword()
+    {
+        const string username = "user1";
+        const string password = "password123";
+        
+        var testSubject = CreateTestSubject(out var credentialStoreMock);
+        credentialStoreMock.Setup(x => x.ReadCredentials(It.Is((TargetUri uri) => UriEquals(uri, SonarQubeUri)))).Returns(new Credential(username, password));
+
+        var response = await testSubject.GetCredentialsAsync(new GetCredentialsParams(SonarQubeConnectionId));
+
+        response.Should().BeEquivalentTo(new GetCredentialsResponse(new UsernamePasswordDto(username, password)));
+    }
+
+    [TestMethod]
+    public async Task GetCredentialsAsync_SonarQubeTokenFound_ReturnsToken()
+    {
+        const string token = "token123";
+        
+        var testSubject = CreateTestSubject(out var credentialStoreMock);
+        credentialStoreMock.Setup(x => x.ReadCredentials(It.Is((TargetUri uri) => UriEquals(uri, SonarQubeUri)))).Returns(new Credential(token));
+
+        var response = await testSubject.GetCredentialsAsync(new GetCredentialsParams(SonarQubeConnectionId));
+
+        response.Should().BeEquivalentTo(new GetCredentialsResponse(new TokenDto(token)));
+    }
+    
+    [TestMethod]
+    public async Task GetCredentialsAsync_SonarCloudUsernameAndPasswordFound_ReturnsUsernameAndPassword()
+    {
+        const string username = "user1";
+        const string password = "password123";
+        
+        var testSubject = CreateTestSubject(out var credentialStoreMock);
+        credentialStoreMock.Setup(x => x.ReadCredentials(It.Is((TargetUri uri) => UriEquals(uri, SonarCloudUri)))).Returns(new Credential(username, password));
```

Comment:
you can put SonarCloudUri as a method parameter without using It.Is for setup which would make it simpler. (no need to change)